vanilla/src/features/films/displayFilmsTable.ts

```diff
@@ -20,106 +27,165 @@ import { renderFilms } from './renderFilms';
+    if (options.valueSearch === '') {
+      this.newValueSearch = null;
+    }
     const filmsTableBody = document.querySelector('.films-table-body');
 
+    this.queryConstraintParameters = {
+      orderingField: this.orderingField,
+      orderingMode: this.orderingMode,
+      valueSearch: this.newValueSearch,
+    };
+
     if (filmsTableBody !== null) {
       try {
-        if (mode === PaginationModes.Init) {
-          films = await FilmsService.fetchFirstPageOfFilms(orderingField, orderingMode);
-        } else if (mode === PaginationModes.Next && !onLastPage) {
-          films = await FilmsService.fetchNextPageOfFilms(lastFilm, orderingField, orderingMode);
-        } else if (!onFirstPage) {
-          films = await FilmsService.fetchPrevPageOfFilms(firstFilm, orderingField, orderingMode);
+        if (options.mode === PaginationModes.Init) {
+          films = await FilmsService.fetchFirstPageOfFilms(this.queryConstraintParameters);
+        } else if (options.mode === PaginationModes.Next && !this.onLastPage && this.lastFilm) {
+          films = await FilmsService.fetchNextPageOfFilms(this.lastFilm, this.queryConstraintParameters);
+        } else if (!this.onFirstPage && this.firstFilm) {
+          films = await FilmsService.fetchPrevPageOfFilms(this.firstFilm, this.queryConstraintParameters);
         }
       } catch (error: unknown) {
-        inProgress = false;
+        this.inProgress = false;
         release();
       }
 
-      if (films.length !== 0) {
-        lastFilm = films[films.length - 1];
-        firstFilm = films[0];
+      ({
+        lastFilm: this.lastFilm, firstFilm: this.firstFilm, onFirstPage: this.onFirstPage, onLastPage: this.onLastPage,
+      } = await this.getPageParameters(
+        films, this.orderingField, this.orderingMode, this.newValueSearch,
+      ));
 
-        onFirstPage = await isFirstPage(firstFilm, orderingField, orderingMode);
-        onLastPage = await isLastPage(lastFilm, orderingField, orderingMode);
+      updatePaginationButtons(this.onFirstPage, this.onLastPage);
 
-        updatePaginationButtons(onFirstPage, onLastPage);
+      updateFieldHeaders(this.orderingField, this.orderingMode);
 
-        updateFieldHeaders(orderingField, orderingMode);
+      renderFilms(films);
 
-        renderFilms(films);
-      }
-
-      inProgress = false;
+      this.inProgress = false;
       release();
     }
-  };
-};
+  }
+
+  /**
+   * Returns the first and last films on the page and whether the given page is the first or last.
+   * @param films Array of films per page.
+   * @param orderingField The field by which the movies are sorted.
+   * @param orderingMode Shows if ordering should be ascending or descending.
+   * @param newValueSearch The actual search value entered by the user..
+   */
+  public static async getPageParameters(films: readonly Film[], orderingField: OrderingFields,
+    orderingMode: OrderingModes, newValueSearch: string | null): Promise<PageParameters> {
```

Comment:
For functions with many arguments it is better to place each of them on a separate line:

```suggestion
  public static async getPageParameters(
  	films: readonly Film[],
  	orderingField: OrderingFields,
    orderingMode: OrderingModes,
    newValueSearch: string | null
  ): Promise<PageParameters> {
```